packages/components/filters/filters-list/src/filters-list.stories.tsx

```diff
@@ -1,22 +1,137 @@
+import { useState } from 'react';
+import { MenuProps, MenuListProps } from 'react-select';
 import FiltersList from './filters-list';
+import SelectInput from '../../../inputs/select-input';
 
 const meta: Meta<typeof FiltersList> = {
   title: 'components/Filters/FiltersList',
   component: FiltersList,
   tags: ['local-dev'],
-  argTypes: {
-    label: {
-      control: 'text',
-    },
-  },
+  argTypes: {},
 };
 export default meta;
 
-type Story = StoryObj<typeof FiltersList>;
+type Story = StoryFn<typeof FiltersList>;
+
+const CustomSelectMenu = ({
+  children,
+  innerProps: { ref, ...restInnerProps },
+}: MenuProps) => (
+  <div ref={ref} {...restInnerProps}>
+    {children}
+  </div>
+);
+
+const CustomMenuList = ({
+  children,
+  innerProps: { ref, ...restInnerProps },
+}: MenuListProps) => (
+  <div ref={ref} {...restInnerProps}>
+    {children}
+  </div>
+);
+
+const primaryColorOptions = [
+  { label: 'Blue', value: 'blue', key: 'blue', id: '2' },
+  { label: 'Red', value: 'red', key: 'red', id: '4' },
+  { label: 'Orange', value: 'orange', key: 'orange', id: '5' },
+  { label: 'Yellow', value: 'yellow', key: 'yellow', id: '6' },
+  { label: 'Green', value: 'green', key: 'green', id: '7' },
+];
+
+const secondaryColorOptions = [
+  { label: 'Purple', value: 'purple', key: 'purple', id: '3' },
+  { label: 'Forest', value: 'forest', key: 'forest', id: '8' },
+  { label: 'Slate', value: 'slate', key: 'slate', id: '9' },
+  { label: 'Silver', value: 'silver', key: 'silver', id: '10' },
+];
+
+export const BasicExample: Story = () => {
+  const [primaryColorValue, setPrimaryColorValue] = useState<string[]>([]);
+  const [secondaryColorValue, setSecondaryColorValue] = useState<string[]>([]);
+
+  const clearPrimaryColorFilter = () => setPrimaryColorValue([]);
+  const clearSecondaryColorFilter = () => setSecondaryColorValue([]);
+
+  const clearAllFilters = () => {
+    clearPrimaryColorFilter();
+    clearSecondaryColorFilter();
+  };
+
+  const appliedFilters = [];
+
+  if (primaryColorValue.length > 0) {
+    appliedFilters.push({
+      filterKey: 'primaryColors',
+      label: primaryColorValue,
+    });
+  }
+
+  if (secondaryColorValue.length > 0) {
+    appliedFilters.push({
+      filterKey: 'secondaryColors',
+      label: secondaryColorValue,
+    });
+  }
+  const filters = [
+    {
+      key: 'primaryColors',
+      label: 'Primary Colors',
+      filter: (
+        <SelectInput
+          name="primaryColorsSelect"
+          options={primaryColorOptions}
+          value={primaryColorValue}
+          onChange={(e) =>
+            setPrimaryColorValue(
+              Array.prototype.concat(e.target.value ? e.target.value : [])
+            )
+          }
+          components={{ Menu: CustomSelectMenu, MenuList: CustomMenuList }}
+          menuIsOpen={true}
+          controlShouldRenderValue={false}
+        />
+      ),
+      onClearRequest: clearPrimaryColorFilter,
+    },
```

Comment:
- This is essentially the config/settings for a single `FilterMenu`, right? What about a `getTags` function as some sort of reducer, which, when executed, returns an array of Tags to be displayed in the underlying FilterMenu. This way any kind of state could be brought into a shape that the FilterMenu understands.

- Should props that are forwarded to the FilterMenu be grouped under a parent key (e.g. `filterMenuConfig`)

- `filter` is so ambiguous/generic, maybe smth. like `filterComponent` or `filterBody` instead? And for performance reasons `filter` (or whatever the new name will be) should be a function returning a React component (otherwise the code is evaluated on every render, regardless of whether this filter is currently displayed or not).

Reply:
> filter is so ambiguous/generic, maybe smth. like filterComponent or filterBody instead? And for performance reasons filter (or whatever the new name will be) should be a function returning a React component (otherwise the code is evaluated on every render, regardless of whether this filter is currently displayed or not).

Going through the document Sarah put together, I noticed the "filter" word appears a lot in the naming, in some cases, almost duplicate naming in different contexts though, is it worth the time to discuss naming patterns regarding filters ?

Reply:
> What about a getTags function as some sort of reducer, which, when executed, returns an array of Tags to be displayed in the underlying FilterMenu.

@misama-ct The only drawback that I can see to this approach is that it might not be straightforward to implement if the filter has an `onApply` action

> Should props that are forwarded to the FilterMenu be grouped under a parent key (e.g. filterMenuConfig)

something like this would work for me (pseudocoded): 
```
{
  key
  label
  filterMenuConfig: {
    menuBody
    onCloseRequest
    ...rest
  }
  ...rest
 }
```

> And for performance reasons filter (or whatever the new name will be) should be a function returning a React component (otherwise the code is evaluated on every render, regardless of whether this filter is currently displayed or not).

This is a good point, in that case should the prop for the input/body start with `render-` e.g. `renderMenuBody`